AsyncDisplayKit/ASMapNode.mm

```diff
@@ -14,11 +14,13 @@
 #import <AsyncDisplayKit/ASCenterLayoutSpec.h>
 #import <AsyncDisplayKit/ASThread.h>
 #import <AsyncDisplayKit/ASInternalHelpers.h>
+#import <AsyncDisplayKit/ASLayout.h>
 
 @interface ASMapNode()
 {
   ASDN::RecursiveMutex _propertyLock;
   MKMapSnapshotter *_snapshotter;
+  BOOL _snapshotNeeded;
```

Comment:
This seems like a good suggestion. Could you recommend a way to replace this? I am not very familiar with this class, but I assume this might involve if the snapshotter is loading or if self.image is set? How should these be combined and is it relevant to check `isLiveMap`?